File: LeetCode/투 포인터 3 마법의 문자열.cpp
class Solution {
public:
    int magicalString(int n) {
        string magic = "";
        for (int i = 1; i <= n; i++) {
            if (i % 2 == 0) {
                magic += "2";
                if (magic[i-1] == '1') {
                    continue;
                } else if (magic[i-1] == '2') {
                    magic += "2";
                }
            } else if (i % 2 == 1) {
                magic += "1";
                if (magic[i-1] == '1') {
                    continue;
                } else if (magic[i-1] == '2') {
                    magic += "1";
                }
            }
        }
        int ans = 0;
        for (int i = 0; i < n; i++) {
            if (magic[i] == '1') {
                ans++;
            }
        }
        return ans;
    }
};

File: LeetCode/투 포인터 3 마법의 문자열.py
class Solution:
    def magicalString(self, n: int) -> int:
        magic = ""
        for i in range(1, n+1):
            if i % 2 == 0:
                magic += "2"
                if magic[i-1] == "1":
                    continue
                elif magic[i-1] == "2":
                    magic += "2"
            elif i % 2 == 1:
                magic += "1"
                if magic[i-1] == "1":
                    continue
                elif magic[i-1] == "2":
                    magic += "1"
        sum = 0
        for i in magic[:n]:
            print(i)
            if i == '1':
                sum += 1
        print(magic[:6])
        return sum
